v1: define names for unknown/default iota values

mgfcmartinez opened the pull request:
v1: define names for unknown/default iota values

Previously we were using "_", and this wasn't a problem because
protoc-gen-go prefaces enums with the enum name to prevent clashes.
However, this is not the case with other languages (specifically Java).
Give actual names for our unknown/default enum values to prevent protoc
errors during compilation.

While at it, get rid of the ISO 4217 currency enum, as the codes change
frequently and the enum is likely to go out of date.

mgfcmartinez commented:
definition changes are [here](https://github.com/openbank/openbank/pull/14/commits/a71d7c9c82d03a64bb0969f5f1e7d4e5b120ba06), the generated files are in a different commit for ease of review. will squash and rebase once approved

nkristianto reviewed (approved):
Thanks for add unknown type for the definition

Kunde21 commented on v1/types/types.gunk:
```diff
@@ -195,508 +195,31 @@ const (
 
 const (
-	_ CardAccessStatus = iota
+	Unkown CardAccessStatus = iota
```
Add the full type name here.

Kunde21 reviewed (approved):
One minor nit

nkristianto commented on v1/types/types.gunk:
```diff
@@ -195,508 +195,31 @@ const (
 	Transferwise
 )
 
-// Currency following format specified on ISO 4217
```
btw miggy, why const of currency is deleted ?

ghazninattarshah replied:
@nkristianto the enum *CurrencyISO4217* is used in ob-services are referred from `transfer` package and not from types. 
@mgfcmartinez CMIIW

Commit:
v1: define names for unknown/default iota values

Previously we were using "_", and this wasn't a problem because
protoc-gen-go prefaces enums with the enum name to prevent clashes.
However, this is not the case with other languages (specifically Java).
Give actual names for our unknown/default enum values to prevent protoc
errors during compilation.

While at it, get rid of the ISO 4217 currency enum, as the codes change
frequently and the enum is likely to go out of date.
